src/Parser.y

```diff
@@ -180,6 +180,11 @@ Top :: { Toplevel Parsed }
     | Access type BindName ListE(TyConArg) TypeBody { withPos2 $2 $3 $ TypeDecl $1 (getL $3) $4 $5 }
     | Access type TyConArg BindOp TyConArg TypeBody { withPos2 $2 $4 $ TypeDecl $1 (getL $4) [$3, $5] $6 }
 
+    | Access type BindName ListE(TyConArg) '<-' Type
```

Comment:
I assume the `<-` is to avoid conflicts with `type t = Foo`? I don't know if it's possible to have `type t = Foo` be a constructor, and `type t = (Foo)` be an alias, or if that's just going to be more confusing...

Reply:
This is a parser decision so I decided to leave it up to you.